### PR TITLE
added Spotify Communicator addon to list of addons in get.php

### PR DESCRIPTION
**Pull request by brlkid**
### Release information
- Name (Example, Add-on Updater): Spotify Communicator
- Author (example, Your Name): Derrick Day <brlkid@outlook.com>
- Repo (example, https://github.com/yourname/addonName): https://github.com/brlkid/Spotify-Communicator-for-NVDA
- Version (example, 21.07): 1.0
- Update channel (example, stable, dev or stable and dev): stable
- NVDA compatibility (example, 2021.1 and beyond): 2020.1 and beyond 

### Changelog (mention changes in separate lines starting with dash space)
This is my initial Pull Request
This addon supplies the ability to announce the currently playing Spotify track with NVDA

### CODE REVIEW

**Review by nvdaes — CHANGES_REQUESTED** [review]
Hello @brlkid: 

Thanks for your pull request to make available your add-on on the website.
I have some suggestions:

- You may add your add-on, with sc id) in alphabetical order. Seems there are add-ons unordered, but I think it's better to add it in a predictable order, so it's easier to ensure that we don't repeat ids in get.php.
- - You're using NVDA+t in a global plugin. This keystroke is used in NVDA's core to retrieve any window title, so this makes difficult to read the title of all windows. I strongly encourage you to use a different keystroke. I don't know if you can create an app module instead of a global plugin, but the keystroke should be changed.
- Optionally, you may also include comments for translators and assign an add-on name equal to your repo, to prevent potential issues in case you decide to get your add-on registered in the NVDA's translation system:
- 
https://github.com/nvaccess/mrconfig

**Comment by brlkid** [comment]
I changed the keystroke because I agree that a conflict would be dificault for users. I also added to the readme to make it more clear what is taking place. Finally, I didn't add translations because no text is hardcoded into the addon so if someone speaks or uses a different language, the titles and artists of the Spotify songs they are listening to should be in the new language. This is my first time doing this so thank you very much for the feedback. It helps me a ton to make not only this addon better, but all of the ones I plan to write in the future!

**Comment by nvdaes** [comment]
Hello, hope this helps you.
Comments for translators can be added for command descriptions or other informational messages. Also, since it's your first time, I'd like to ask you: wouldn't be better to use an app module instead of a global plugin? I don't have Spotify here and I haven't tried to use an app module. But if the command works just when Spotify is active, you can consider to use an app module instead of fetching for children of the desktop object.
Let us if you want to make more changes before we approve this pull request.
Cheers.

**Comment by brlkid** [comment]
OK. I'll go in and add Translations for the command gesture descriptions. The reason I went with a global plug-in is because the Spotify app has to be open So music is playing although the whole point of the Alan is so you can access the title of the currently playing Spotify track when you're in other applications. For example, if you have music playing and your editing a Microsoft Word document. I will definitely add the descriptions. Is there anything else that I need to do to allow my ad on to get approved? Do I need to go and add my adon anywhere else in any other database or file? Thank you so much, again, I really am new to this and don't know a lot about what I'm doing but your help has been awesome!

**Comment by nvdaes** [comment]
Hello: the only requirement to get an add-on approved is that manifest.ini and the download url are right, but I made a quick review to provide feedback in case you want to improve it. I"ll wait until the weekend to give more time in case we want to discuss some other points before approving it.

Enviado desde mi iPhone

**Comment by nvdaes** [comment]
Hi again:
I've reviewed the add-on source code again. I have more suggestions since it's your first time:

- You may include a category for your scripts, so it's easier to organize them in the Input gestures dialog.
- The readme may contain headings at level 2 to introduce commands if you want
- In the readme available at the main branch of your repo, the current gesture is not updated.
- I'm not sure if the translation system, if you want to register this add-on, support spaces in names. camel case or underscores are supported.
- NVDA+shift+t can be used for your add-on. Anyway, it's used for a popular add-on which is instantTranslate.
- You may subscribe to the add-ons mailing list to get more feedback. It's a mailing list created by @josephsl available at
https://nvda-addons.groups.io/g/nvda-addons

Hope this helps.

**Comment by brlkid** [comment]
I will make those changes. Thank you so much, I know that all of the reviewing of all of my files takes a while and it is awesome that you are doing all that as a volunteer. I was trying to figure out how to make the headings actually marked as headings but I had to use markdown rather than html and I couldn’t find an accessible markdown editor. I will keep doing more looking for that. The readme also does now reflect the accurate keystroke as well, thank you for noticing that. I did see that conflict with the other addon however I don’t know of another easily pressable keystroke that isn’t a conflict with anything. I will definetly make a Spotify Communicator category so that way the keystroke in the case that the person has both addons installed, can e easily changed. I will push all of my changes so you can look at them and I look forward to hearing from you soon.
P.S. I will also join the addon mailing list so I can get this feedback before I go to publish in the future.

Thanks,
Derrick C. Day

**Comment by nvdaes** [comment]
Thank you.
For a markdown editor, you may use VS Code, and press control+shift+v to preview the markdown like it appears in a web browser. I often use Notepad++ too.
Hope this helps.

File: get.php
Comment (nvdaes) [inline get.php:124]:
```suggestion
	"sc" => "https://github.com/brlkid/Spotify-Communicator-for-NVDA/releases/download/v1.0/Spotify.Communicator-1.0.nvda-addon",
```

**Review by nvdaes — CHANGES_REQUESTED** [review]

**Comment by nvdaes** [comment]
Hi again, I'll wait for your comments before approving this PR. Anyway this line needs to be changed in get.php: It's just a comma.
Also, usually we add license and copyright headers in Python modules, though your repo contains a text file with license.
You can see other add-ons if you need examples.
I can show you an add-on created by me for simplicity. But there are lots of add-ons to see.
I think this is a super single add-on,with a very good documentation authored by a native English speaker, not me. He's @GeorgeKerscher
Andthe add-on can be found at

https://github.com/nvdaes/enhancedAnnotations

**Comment by brlkid** [comment]
I have added the comma. Sorry about that, I don’t know why I didn’t catch it before I committed. I haven’t included the licensing headers yet. If that isn’t required, I would say we’re ready but if you need me to add them, it is no problem. Thanks for your help.

Thanks,
Derrick C. Day

**Review by nvdaes — CHANGES_REQUESTED** [review]
Hello, we can approve this if you want, but seems that in the linked version, the gesture is NVDA+t, used to get window titles. I think you want to change it and it's a mistake. Don't worry, and take your  time to fix it when you want.
About headers of the Python file to include the license, I think it's not needed, and in fact I'm not a specialist about license requirements. I just follow conventions in this matter. The help requested to me about this repo is just related to review the manifest and add-on urls to ensure that NVDA security mechanism about compatibility is OK and that the add-on can be downloaded from the provided address.
But I make you suggestions also for your interest and to make your add-on more useful, specially considering that it's your first time.
I'll wait for you.
Also, I noticed that you moved the link to speakPasswords add-on but the link seems to be the same, so it's OK for me to placekeys in alphabetical order.

**Comment by josephsl** [comment]
Hi,

Update: could you mind testing the add-on with NVDA 2023.1 and update the manifest accordingly? Thanks.

**Comment by seanbudd** [comment]
This repository has been archived. Please submit to the new repository. https://github.com/nvaccess/addon-datastore/blob/master/README.md#submitting-an-add-on-version

**Pull request closed without merging**
